File: Backend/Microservices/Expense/EMS.Core/Entities/Media.cs
using EMS.Core.Entities.Common;
using EMS.Core.Enums;

namespace EMS.Core.Entities
{
    public class Media : BaseAuditableEntity<Guid>
    {
        public string FileName { get; set; } = default!;
        public string? ContentType { get; set; } // MIME
        public string? Url { get; set; } = default!;
        public string? SecureUrl { get; set; } = default!;
        public string? ThumbnailUrl { get; set; }
        public long Size { get; set; }
        public string Extension { get; set; } = default!;
        public FileType Type { get; set; }
        public string? PublicId { get; set; }
        public string? AssetId { get; set; }
        public string? AltText { get; set; }
        public string? Caption { get; set; }

        // Dimension
        public int? Width { get; set; }
        public int? Height { get; set; }
        public int? Duration { get; set; }

        // Tracking
        public MediaStatus Status { get; set; } = MediaStatus.Pending;
        //public string? Provider { get; set; }
        //public string? StoragePath { get; set; }
        public string? Metadata { get; set; }
        public bool IsOptimized { get; set; } = false;

        public int? ChatMessageId { get; set; }

        // Navigations
        public ChatMessage? ChatMessage { get; set; }
        public ICollection<Category> Categories { get; set; } = [];
    }
}

File: Clients/mobile/expense_management_system/lib/feature/chat/model/media.dart
// media.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'media.freezed.dart';
part 'media.g.dart';

@freezed
class Media with _$Media {
  const factory Media({
    required String id,
    required String url,
    required String secureUrl,
    String? thumbnailUrl,
    required int size,
    required String extension,
    required String type,
    required String publicId,
    required String assetId,
    String? altText,
    String? caption,
    int? width,
    int? height,
    int? duration,
    required String status,
    required String metadata,
    required bool isOptimized,
    required int chatMessageId,
    required DateTime createdAt,
  }) = _Media;

  factory Media.fromJson(Map<String, dynamic> json) => _$MediaFromJson(json);
}
